nepomuk/src/main/java/edu/asu/diging/gilesecosystem/nepomuk/core/service/impl/RequestProcessor.java

```diff
@@ -126,6 +138,34 @@ public void processRequest(IStorageRequest request) {
         }
     }
     
+    @Override
+    public void processRequest(IStorageDeletionRequest request) {
+        IFile file = filesManager.getFile(request.getStorageFileId());
+        IFileTypeHandler handler = fileHandlerRegistry.getHandler(file.getFileType());
+        handler.deleteFile(file, request.getIsOldFileVersion());
+        List<IFile> files = filesManager.getFilesByDocumentId(file.getDocumentId());
+        // If no files are remaining for the document the request to delete the document and upload in Giles is pushed. 
+        if (files.isEmpty()) {
```

Comment:
I don't understand, what is this check for?

Reply:
The Document and Upload in Giles will be deleted once all the files associated with them are deleted in nepomuk. Thus I'm checking if for a given document there are no files associated before sending a CompletedStorageDeletionRequest to giles to delete the document and upload

Reply:
there should probably be just one request to delete multiple files, not multiple requests. otherwise, Giles never knows what the status of the individual requests is.

Reply:
also, with multiple nepomuk instances, this can lead to multiple completion requests.